### ISSUE TITLE
Menu cart widget change in behaviour

### ISSUE THREAD

**Issue by lewclarke** [elementor/elementor#9336]
## Prerequisites

- [x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [x] The issue still exists against the latest stable version of Elementor.

## Description

On my website I use the menu cart widget, but I have never wanted the menu "slideover" when you click the cart icon, I have always wanted it to go straight to the cart. I have achieved this for a very long time, using the following line of code

```
add_filter( 'woocommerce_widget_cart_is_hidden', '__return_true' );
```
This has always had the desired effect, clicking the menu cart would not show the slideover and go straight to the cart. I have been running elementor pro 2.6.5 with no issue. Today I upgraded to 2.7.2 and this no longer works. With this piece of code, everything is now hidden, not just the slideover, but the actual cart button itself. I have looked through the changelog between these 2 versions and can see nothing mentioning change in behaviour to this widget, so was surprised to see this happen. Right now I have had to rollback to 2.6.5 and cannot upgrade as this breaks key functionality. My question is, why has this behaviour changed? Is this an intended change in behavior that wasn't documented, or is this a bug in that the widget no longer respects the woocommerce filter as it did?

## Steps to reproduce

1. Upgrade to Elementor 2.7.4/Elementor Pro 2.7.2
2. Filter code (as seen above) now hides the cart button, not just the cart slideover. Prior to this version, the cart button would appear, and when clicked go straight to the cart. 

## Isolating the problem

- [x] This bug happens with only Elementor plugin active (and Elementor Pro).
- [ ] This bug happens with a default WordPress theme active.
- [x] I can reproduce this bug consistently using the steps above.

**Comment by lewclarke** [comment]
I've spent some time digging into the code for the menu cart widget between elementor pro 2.6.5 and 2.7.2 and can see there has been a major change to the code for this widget. In 2.6.5 if 'woocommerce_widget_cart_is_hidden' was true or you was on the cart/checkout page, the menu slide over would not appear. In 2.7.2, it only hides the menu if your on the cart/checkout page, it no longer checks the 'woocommerce_widget_cart_is_hidden' setting. 

Again, I'm just curious as to why this change was made. In 2.6.5 it was possible to hide the menu everywhere (I do not want the slide over on my site), but in 2.7.2 the code changes made to the widget now make this impossible. Am i missing something here, or is this intentional?

**Comment by shilo-ey** [comment]
Hi @lewclarke 
Try using WP Dashboard > Elementor > Settings > Integrations > WooCommerce > Mini Cart Template - Choose "Disable" and see if that solves the issue for you.

Thanks

**Comment by lewclarke** [comment]
@shilo-ey Sorry for the delay in getting back to you. I did try that setting and confirm it doesn't work. As said in my previous post when digging through the source code for the plugin I can see it no longer supports working in this way (aka use the menu cart widget without the minicart). I have updated to the latest versions today and confirm its still the same. Its a shame that this functionality was taken away, it was nice to be able to use this widget without the mini cart, so for me I can no longer use the widget on my site since 2.7.2.
